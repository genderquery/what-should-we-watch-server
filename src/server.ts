import http from "http";
import { AddressInfo } from "net";
import app from "./app";

const server = http.createServer(app);

server.listen(process.env.PORT || 5000, () => {
  let { address, port } = server.address() as AddressInfo;
  if (address === "" || address === "::") {
    address = "localhost";
  }
  console.log(`Listening at http://${address}:${port}`);
});
