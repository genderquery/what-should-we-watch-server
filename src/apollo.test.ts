import { gql } from "apollo-server-express";
import { createTestClient } from "apollo-server-testing";
import server from "./apollo";

test("hello", async () => {
  const { query } = createTestClient(server);
  const result = await query({
    query: gql`
      query HelloQuery {
        hello
      }
    `,
  });
  expect(result).toMatchSnapshot();
});
