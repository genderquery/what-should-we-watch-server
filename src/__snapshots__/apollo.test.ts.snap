// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hello 1`] = `
Object {
  "data": Object {
    "hello": "Hello world!",
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
